File: 2537_count_the_number_of_good_subarrays.cpp
class Solution {
public:
    long long countGood(vector<int>& a, int b) {
        unordered_map<int, int> m; long long s = 0, c = 0;
        for (int e = 0, h = 0; h < a.size(); ++h) {
            c += m[a[h]]++;
            while (c >= b) c -= --m[a[e++]];
            s += e;
        }
        return s;
    }
};

File: 2537_count_the_number_of_good_subarrays.java
class Solution {
    public long countGood(int[] x, int y) {
        Map<Integer, Integer> m = new HashMap<>();
        long r = 0, c = 0;
        for (int i = 0, j = 0; j < x.length; ++j) {
            c += m.getOrDefault(x[j], 0);
            m.put(x[j], m.getOrDefault(x[j], 0) + 1);
            while (c >= y) {
                m.put(x[i], m.get(x[i]) - 1);
                c -= m.get(x[i++]);
            }
            r += i;
        }
        return r;
    }
}
